labextension/src/restapi.ts

```diff
@@ -0,0 +1,58 @@
+import { URLExt } from "@jupyterlab/coreutils";
+import { ServerConnection } from "@jupyterlab/services";
+import { IModel } from "./serverproxy";
+
+/**
+ * The url for the server proxy service.
+ */
+const SERVER_PROXY_SERVICE_URL = "api/server-proxy";
+
+/**
+ * List the running server proxy apps.
+ *
+ * @param settings - The server settings to use.
+ *
+ * @returns A promise that resolves with the list of running session models.
+ */
+export async function listRunning(
+  settings: ServerConnection.ISettings = ServerConnection.makeSettings(),
```

Comment:
having the manager do this (with exactly one `settings`, handed in at startup) would make these signatures a lot simpler.